lib/rules/no-undef-init.js

```diff
@@ -34,37 +34,88 @@ module.exports = {
 
         const sourceCode = context.getSourceCode();
 
+        /**
+         * Get the node of init target.
+         * @param {ASTNode} node The node to get.
+         * @returns {ASTNode} The node of init target.
+         */
+        function getIdNode(node) {
+            switch (node.type) {
+                case "VariableDeclarator":
+                    return node.id;
+                case "PropertyDefinition":
+                    return node.key;
+                default:
+                    throw new Error("unreachable");
+            }
+        }
+
+        /**
+         * Get the node of init value.
+         * @param {ASTNode} node The node to get.
+         * @returns {ASTNode} The node of init value.
+         */
+        function getInitNode(node) {
+            switch (node.type) {
+                case "VariableDeclarator":
+                    return node.init;
+                case "PropertyDefinition":
+                    return node.value;
+                default:
+                    throw new Error("unreachable");
+            }
+        }
+
+        /**
+         * Get the parent kind of the node.
+         * @param {ASTNode} node The node to get.
+         * @returns {string} The parent kind.
+         */
+        function getParentKind(node) {
+            switch (node.type) {
+                case "VariableDeclarator":
+                    return node.parent.kind;
+                case "PropertyDefinition":
+                    return "field";
+                default:
+                    throw new Error("unreachable");
+            }
+        }
+
         return {
 
-            VariableDeclarator(node) {
-                const name = sourceCode.getText(node.id),
-                    init = node.init && node.init.name,
+            "VariableDeclarator, PropertyDefinition"(node) {
```

Comment:
I think `PropertyDefinition` should be behind an option (or a separate rule?), the docs for this rule says that it checks variable declarations.